website/docs/r/network_connection_monitor.html.markdown

```diff
@@ -10,109 +10,282 @@ description: |-
+
+* `filter` - (Optional) A `filter` block as defined below.
+
+* `virtual_machine_id` - (Optional) The ID of the Virtual Machine which is used as the endpoint by the Network Connection Monitor.
+
+---
 
-* `port` - (Required) The destination port used by connection monitor.
+A `filter` block supports the following:
 
-* `address` - (Optional) The address of the connection monitor destination (IP or domain name). Conflicts with `destination.0.virtual_machine_id`
+* `type` - (Optional) The behavior type of this endpoint filter. Currently the only allowed value is `Include`. Defaults to `Include`.
 
-* `virtual_machine_id` - (Optional) The ID of the virtual machine used as the destination by connection monitor. Conflicts with `destination.0.address`
+* `item` - (Optional) A `item` block as defined below.
 
 ---
 
-A `source` block supports the following:
+A `item` block supports the following:
+
+* `type` - (Optional) The type of items included in the filter. Possible values are `AgentAddress`. Defaults to `AgentAddress`.
+
+* `address` - (Optional) The address of the filter item.
+
+---
+
+A `test_configuration` block supports the following:
+
+* `name` - (Required) The name of test configuration for the Network Connection Monitor.
+
+* `protocol` - (Required) The protocol used to evaluate tests. Possible values are `Tcp`, `Http` and `Icmp`.
+
+* `test_frequency_in_seconds` - (Optional) The time interval in seconds at which the test evaluation will happen. Defaults to `60`.
+
+* `http_configuration` - (Optional) A `http_configuration` block as defined below.
+
+* `icmp_configuration` - (Optional) A `icmp_configuration` block as defined below.
+
+* `preferred_ip_version` - (Optional) The preferred IP version which is used in test evaluation. Possible values are `IPv4` and `IPv6`. 
+
+* `success_threshold` - (Optional) A `success_threshold` block as defined below.
+
+* `tcp_configuration` - (Optional) A `tcp_configuration` block as defined below.
+
+---
+
+A `http_configuration` block supports the following:
+
+* `method` - (Optional) The HTTP method for the HTTP request. Possible values are `Get` and `Post`. Defaults to `Get`.
+
+* `port` - (Optional) The port for the Http connection.
+
+* `path` - (Optional) The path component of the URI. For instance, `/dir1/dir2`.
+
+* `prefer_https` - (Optional) Should https be preferred over http in cases where the choice is not explicit? Defaults to `false`.
+
+* `request_header` - (Optional) A `request_header` block as defined below.
+
+* `valid_status_code_ranges` - (Optional) The http status codes to consider successful. For instance, `2xx`, `301-304` and `418`.
+
+---
+
+A `request_header` block supports the following:
+
+* `name` - (Required) The name of the HTTP header.
+
+* `value` - (Required) The value of the HTTP header.
+
+---
+
+A `icmp_configuration` block supports the following:
+
+* `trace_route_disabled` - (Optional) Should path evaluation with trace route be disabled? Defaults to `false`.
```

Comment:
Can we negate this to be `trace_route_enabled`, and change the logic to reflect this? Since it is more intuitive for users.

Reply:
updated